src/solver/tron.jl

```diff
@@ -0,0 +1,403 @@
+#  Some parts of this code were adapted from
+# https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/optimize/tron.py
+
+using LinearOperators, NLPModels
+
+export tron, project!, project_step!, active, breakpoints
```

Comment:
Do we want all that suff exported, or just `tron`? Perhaps the others could be a general utilities submodule?!